pages/editor.js

```diff
@@ -23,51 +24,64 @@ import {
   COLORS,
   DEFAULT_CODE
 } from '../lib/constants'
+import { mappings, reverseMappings, replaceQuery } from '../lib/routing'
 import { getState, saveState } from '../lib/util'
 
+const mapper = new Morph()
+
 class Editor extends React.Component {
-  /* pathname, asPath, err, req, res */
-  static async getInitialProps({ asPath }) {
+  static async getInitialProps({ asPath, query }) {
+    const queryParams = mapper.map(mappings, query)
+    const initialState = Object.keys(queryParams).length ? queryParams : null
     try {
       // TODO fix this hack
       if (asPath.length > 30) {
         const content = await api.getGist(asPath)
-        return { content }
+        return { content, intialState }
       }
     } catch (e) {
       console.log(e)
     }
-    return {}
+    return { initialState }
   }
 
   constructor(props) {
     super(props)
-    this.state = {
-      background: 'rgba(171, 184, 195, 1)',
-      theme: THEMES.seti.id,
-      language: DEFAULT_LANGUAGE,
-      dropShadow: true,
-      windowControls: true,
-      widthAdjustment: true,
-      paddingVertical: '48px',
-      paddingHorizontal: '32px',
-      uploading: false,
-      code: props.content
-    }
+    this.state = Object.assign(
+      {
+        background: 'rgba(171, 184, 195, 1)',
+        theme: THEMES.seti.id,
+        language: DEFAULT_LANGUAGE,
+        dropShadow: true,
+        windowControls: true,
+        widthAdjustment: true,
+        paddingVertical: '48px',
+        paddingHorizontal: '32px',
+        uploading: false,
+        code: props.content,
+        _initialState: this.props.intialState
```

Comment:
Typo: should probably be `initialState`. L62 and L64.

Reply:
Great catch! I might have never seen that!